docs/source/nbs/schema.ipynb

```diff
@@ -0,0 +1,328 @@
+       "   'minProperties': {'$ref': '#/definitions/nonNegativeIntegerDefault0'},\n",
+       "   'required': {'$ref': '#/definitions/stringArray'},\n",
+       "   'additionalProperties': {'$ref': '#'},\n",
+       "   'definitions': {'type': 'object',\n",
+       "    'additionalProperties': {'$ref': '#'},\n",
+       "    'default': {}},\n",
+       "   'properties': {'type': 'object',\n",
+       "    'additionalProperties': {'$ref': '#'},\n",
+       "    'default': {}},\n",
+       "   'patternProperties': {'type': 'object',\n",
+       "    'additionalProperties': {'$ref': '#'},\n",
+       "    'propertyNames': {'format': 'regex'},\n",
+       "    'default': {}},\n",
+       "   'dependencies': {'type': 'object',\n",
+       "    'additionalProperties': {'anyOf': [{'$ref': '#'},\n",
+       "      {'$ref': '#/definitions/stringArray'}]}},\n",
+       "   'propertyNames': {'$ref': '#'},\n",
+       "   'const': True,\n",
+       "   'enum': {'type': 'array',\n",
+       "    'items': True,\n",
+       "    'minItems': 1,\n",
+       "    'uniqueItems': True},\n",
+       "   'type': {'anyOf': [{'$ref': '#/definitions/simpleTypes'},\n",
+       "     {'type': 'array',\n",
+       "      'items': {'$ref': '#/definitions/simpleTypes'},\n",
+       "      'minItems': 1,\n",
+       "      'uniqueItems': True}]},\n",
+       "   'format': {'type': 'string'},\n",
+       "   'contentMediaType': {'type': 'string'},\n",
+       "   'contentEncoding': {'type': 'string'},\n",
+       "   'if': {'$ref': '#'},\n",
+       "   'then': {'$ref': '#'},\n",
+       "   'else': {'$ref': '#'},\n",
+       "   'allOf': {'$ref': '#/definitions/schemaArray'},\n",
+       "   'anyOf': {'$ref': '#/definitions/schemaArray'},\n",
+       "   'oneOf': {'$ref': '#/definitions/schemaArray'},\n",
+       "   'not': {'$ref': '#'}},\n",
+       "  'default': True},\n",
+       " 'http://json-schema.org/draft-07/schema')"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "a.schema = \"http://json-schema.org/draft-07/schema\"\n",
+    "a.schema, a.schema_source"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## From a private repo"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For github, you just specify the raw link which will contain a token on the end\n",
+    "\n",
+    "```a.schema = \"https://raw.githubusercontent.com/manycoding/test_schema/master/deep/schema.json?token=AJ6jjTtZtWZr5zyw7DuWduieMJ2ms1ks5ctRC6wA%3%3D\"```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For bitbucket, you have to set up `BITBUCKET_USER` and `BITBUCKET_PASSWORD` environment variables.\n",
+    "For example, in Jupyter it looks like:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
```

Comment:
What do you think for adding an option for JSON or YAML config file.

Reply:
Nobody really requested this so far.